pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -309,14 +310,35 @@ func findContainerIDByName(status *v1.PodStatus, name string) (string, error) {
 	return "", fmt.Errorf("unable to find ID for container with name %v in pod status (it may not be running)", name)
 }
 
-func (m *manager) updateContainerCPUSet(containerID string, cpus cpuset.CPUSet) error {
+func (m *manager) updateContainerCPUSet(pod *v1.Pod, container *v1.Container, containerID string, cpus cpuset.CPUSet) error {
 	// TODO: Consider adding a `ResourceConfigForContainer` helper in
 	// helpers_linux.go similar to what exists for pods.
 	// It would be better to pass the full container resources here instead of
 	// this patch-like partial resources.
-	return m.containerRuntime.UpdateContainerResources(
-		containerID,
-		&runtimeapi.LinuxContainerResources{
-			CpusetCpus: cpus.String(),
-		})
+	resources := &runtimeapi.LinuxContainerResources{
+		CpusetCpus: cpus.String(),
+	}
+
+	// When a dedicated CPU set is assigned, unset CFS CPU quota to avoid performance
+	// degradation due to the way this quota is implemented in the Linux kernel.
+	if policyName(m.policy.Name()) != PolicyNone && GuaranteedCPUs(pod, container) != 0 {
```

Comment:
Not crazy about this condition... it does seem like an abstraction leak.

Could you instead add a function like `EnforceCpuQuota(pod *v1.Pod, container *v1.Container) bool` to the Policy interface and then call that from here to determine whether to override the quota?

Reply:
Thanks, that's indeed a much nicer way. Done!